sktime/series_as_features/base/estimators/_ensemble.py

```diff
@@ -0,0 +1,355 @@
+                    .format(estimator.__class__.__name__)
+                ) from exc
+            raise
+    else:
+        estimator.fit(X, y)
+    return estimator
+
+
+def _parallel_build_trees(tree, forest, X, y, sample_weight, tree_idx, n_trees,
+                          verbose=0, class_weight=None,
+                          n_samples_bootstrap=None):
+    """
+    Private function used to fit a single tree in parallel."""
+    if verbose > 1:
+        print("building tree %d of %d" % (tree_idx + 1, n_trees))
+
+    # name of step of final estimator in pipeline
+    estimator = tree.steps[-1][0]
+
+    if forest.bootstrap:
+        n_samples = X.shape[0]
+        if sample_weight is None:
+            curr_sample_weight = np.ones((n_samples,), dtype=np.float64)
+        else:
+            curr_sample_weight = sample_weight.copy()
+
+        indices = _generate_sample_indices(tree.random_state, n_samples,
+                                           n_samples_bootstrap)
+        sample_counts = np.bincount(indices, minlength=n_samples)
+        curr_sample_weight *= sample_counts
+
+        if class_weight == 'subsample':
+            with catch_warnings():
+                simplefilter('ignore', DeprecationWarning)
+                curr_sample_weight *= compute_sample_weight('auto', y, indices)
+        elif class_weight == 'balanced_subsample':
+            curr_sample_weight *= compute_sample_weight('balanced', y, indices)
+        fit_params = {f'{estimator}__sample_weight': curr_sample_weight}
+        tree.fit(X, y, **fit_params)
+    else:
+        fit_params = {f'{estimator}__sample_weight': sample_weight}
+        tree.fit(X, y, **fit_params)
+
+    return tree
+
+
+class BaseTimeSeriesForest(BaseForest):
+    """
+    Base class for forests of trees.
+    """
+
+    @abstractmethod
+    def __init__(self,
+                 base_estimator,
+                 n_estimators=100,
+                 estimator_params=tuple(),
+                 bootstrap=False,
+                 oob_score=False,
+                 n_jobs=None,
+                 random_state=None,
+                 verbose=0,
+                 warm_start=False,
+                 class_weight=None,
+                 max_samples=None):
+        super(BaseTimeSeriesForest, self).__init__(
+            base_estimator,
+            n_estimators=n_estimators,
+            estimator_params=estimator_params
+        )
+
+        self.bootstrap = bootstrap
+        self.oob_score = oob_score
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+        self.verbose = verbose
+        self.warm_start = warm_start
+        self.class_weight = class_weight
+        self.max_samples = max_samples
+
+    @abstractmethod
+    def check_estimator(self):
+        """
+        Checks if the estimator is a Classifier or a Regressor
+        To be implemented by subclasses
+        """
+
+    @abstractmethod
+    def _validate_estimator(self, default=None):
```

Comment:
This can be removed or be an empty function, it's implemented in the concrete classes

Reply:
Yes made it empty function. Removing this causes tests to fail since the check of `base_estimator` fails here whereas since it is abstract that test isn't needed.